### PR TITLE
Add support for regional secrets list datasource `google_secret_manager_regional_secrets`

### PR DESCRIPTION
**Pull request by abheda-crest**
Add support for regional secrets list datasource `google_secret_manager_regional_secrets`.
More info about regional secrets: https://cloud.google.com/secret-manager/docs/regional-secrets-overview

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-datasource
`google_secret_manager_regional_secrets`
```

### CODE REVIEW

**Comment by ScottSuarez** [comment]
Can I ask for my information about your usecase? How are are you looking to use and interact with this resource?

**Comment by abheda-crest** [comment]
> Can I ask for my information about your usecase? How are are you looking to use and interact with this datasource?

This datasource enables the retrieval of regional secrets for a specified location. It is particularly valuable for users who want to filter regional secrets [using specific parameters](https://cloud.google.com/secret-manager/docs/filtering) and leverage these values in other resources (e.g. to fetch secret version based on the name of the filtered regional secrets). Moreover, It is regional equivalent of the datasource [`google_secret_manager_secrets`](https://registry.terraform.io/providers/hashicorp/google/latest/docs/data-sources/secret_manager_secrets).
References: https://github.com/hashicorp/terraform-provider-google/issues/14556 & https://github.com/GoogleCloudPlatform/magic-modules/pull/9099

File: mmv1/third_party/terraform/services/secretmanagerregional/data_source_secret_manager_regional_secrets.go [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+}
+
+func dataSourceSecretManagerRegionalRegionalSecretsRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*transport_tpg.Config)
+	userAgent, err := tpgresource.GenerateUserAgentString(d, config.UserAgent)
+	if err != nil {
+		return err
+	}
+
+	url, err := tpgresource.ReplaceVars(d, config, "{{SecretManagerRegionalBasePath}}projects/{{project}}/locations/{{location}}/secrets")
```
Comment (gptSanyam) [inline mmv1/third_party/terraform/services/secretmanagerregional/data_source_secret_manager_regional_secrets.go:50]: Can we not use the entire path of the resource from config here.
We seem to be using : SecretManagerRegionalBasePath here.
Can we not use the base_url here?

**Review by gptSanyam — APPROVED** [review]
LGTM

File: mmv1/third_party/terraform/services/secretmanagerregional/data_source_secret_manager_regional_secrets.go (hunk 1 above)
Comment (abheda-crest) [reply mmv1/third_party/terraform/services/secretmanagerregional/data_source_secret_manager_regional_secrets.go:50]: Internally, it uses the base_url defined in the Product.yaml. The entry for `SecretManagerRegionalBasePath` is generated in the `transport/config.go` file which be found here: https://github.com/hashicorp/terraform-provider-google/blob/main/google/transport/config.go#L563

**Pull request merged**